client/src/controllers/BulkController.ts

```diff
@@ -0,0 +1,69 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Controller that adds the ability to toggle some or all the elements, ignoring the disabled ones
+ * Additionaly, it checks if all non-disabled items are selected or not. If they are, then select all
+ *
+ * @example
+ * <div data-controller="w-bulk">
+ *   <input type="checkbox" data-w-bulk-target="all">
+ *
+ *   <div>
+ *     <input type="checkbox" data-w-bulk-target="item" disabled>
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *   </div>
+ *
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
+ * </div>
+ */
+export class BulkController extends Controller {
+  /** The list of targets for this controller. */
+  static targets = ['all', 'item'];
+
+  /** The list of all select-all checkbox targets for this controller. */
```

Comment:
```suggestion
  /** All select-all checkbox targets */
```
Shorter is better, we know this is a list beacuse you have `[]` at the end. We also know this is a controller.